src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/Intermediate/ComponentIntermediateNode.cs

```diff
@@ -49,6 +49,8 @@ public sealed class ComponentIntermediateNode : IntermediateNode
 
     public string TypeName { get; set; }
 
+    public SourceSpan? StartTagSpan { get; set; }
```

Comment:
`ComponentIntermediateNode` is only ever created in one place: `ComponentLoweringPass.RewriteAsComponent(...)`. Assuming that's still true after this pull request, consider changing the type of `StartTagSpan` from `SourceSpan?` to `SourceSpan` and make it init-only.